realm/realm-library/src/main/java/io/realm/Mixed.java

```diff
@@ -19,537 +19,379 @@
 import org.bson.types.Decimal128;
 import org.bson.types.ObjectId;
 
-import java.util.Collections;
 import java.util.Date;
 import java.util.UUID;
 
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
-import io.realm.internal.ManageableObject;
-import io.realm.internal.OsSharedRealm;
-import io.realm.internal.RealmProxyMediator;
-import io.realm.internal.Row;
-import io.realm.internal.Table;
-
 
 /**
- * {@link Mixed} is used to represent a polymorphic Realm value.
- * It has two modes: a managed and unmanaged mode. In managed mode contents are persisted inside a Realm, in
- * unmanaged mode contents are persisted in the object instance.
+ * {@link io.realm.Mixed} is used to represent a polymorphic Realm value.
+ * <p>
+ * At any particular moment an instance of this class stores a
+ * definite value of a definite type. If, for instance, that is an
+ * double value, you may call asDouble() to extract that value. You
+ * may call getType() to discover what type of value is currently
+ * stored. Calling asDouble() on an instance that does not store an
+ * integer would raise a {@link java.lang.ClassCastException}.
```

Comment:
```suggestion
 * double would raise a {@link java.lang.ClassCastException}.
```

I assume you mean double here? Integer feels a bit inprecise